File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void){
  
    float fchange;
    int ichange, money=0, i=0, novcici[]={25,10,5,1};
    printf("O hai!");
    do {
        printf("How much change is owed?\n");
        fchange = get_float();
    } while( fchange < 0); 
    ichange = round(fchange * 100);
    while(ichange>0){
        if(ichange-novcici[i]<0){
            i++;
        }
        else{
            ichange-=novcici[i];
            money++;
        }
    }
    printf("%i\n",money);
}

File: pset6/greedy.py
#!/usr/bin/python3

import sys
import cs50
import math

def main():
    novci = [25, 10, 5, 1]
    fchange = -1
    i = 0
    
    print("O hai(o)")
    while(fchange < 0):
        fchange =  float(input("How much change is owed?\n"))
    ichange = round(fchange * 100)
    money = 0
    while(ichange > 0):
        if ichange - novci[i] < 0:
            i = i + 1
        else:
            ichange -= novci[i]
            money = money + 1
    print(money)
            
        
if __name__ == "__main__":
    main()